proposals/shared-everything-threads/Overview.md

```diff
@@ -401,24 +401,26 @@ primitive that does not depend on linear memory. We propose adding a new abstrac
 `waitqueue` to serve as this new primitive.
 
 Unlike other abstract heap types, which are only ever subtypes of other abstract heap types (or are
-bottom types), `waitqueue` is a final subtype of `(rec (type (sub (struct shared (field i32))))).0`,
-meaning it is a struct with one visible `i32` field that can be accessed with all the standard
-struct accessors as well as the new atomic struct accessors. This field is the futex control field
-that is atomically checked when waiting on the waiter queue. `waitqueue` is also always shared.
-There is no non-shared version of it. It is not valid to declare a new subtype of `waitqueue`.
+bottom types), `waitqueue` is a final subtype of `(rec (type (sub (struct shared (field (mut (ref
+null (shared eq)))))))).0`, meaning it is a struct with one visible shared eqref field that can be
+accessed with all the standard struct accessors as well as the new atomic struct accessors. This
+field is the futex control field that is atomically checked when waiting on the waiter queue.
+`waitqueue` is also always shared. There is no non-shared version of it. It is not valid to declare
+a new subtype of `waitqueue`.
```

Comment:
Nit: The last sentence is redundant, since it already says above that it is a final type. (Also, there isn't even syntax to subtype a built-in type.)